python/ql/src/semmle/python/security/TaintTracking.qll

```diff
@@ -1248,10 +1248,42 @@ library module TaintFlowImplementation {
                 test.getSense() = false and not kind.getClass().getAnImproperSuperType() = cls
             )
             or
-            test.getTest() = var.getSourceVariable().getAUse() and kind.booleanValue() = test.getSense()
+            test.getSense() = test_evaluates(test.getTest(), var.getSourceVariable().getAUse(), kind)
         )
     }
 
+    /** Holds if `expr` is the operand of a unary `not` expression. */
```

Comment:
This looks more like ``/** Gets the operand of the unary `not` expression `expr`. */``